Controller/ProfilerController.php

```diff
@@ -85,6 +85,35 @@ public function explainAction($token, $connectionName, $query)
         ));
     }
 
+    public function executeAction($token, $connectionName, $query)
+    {
+        $queryId = $query;
+
+        /** @var $profiler \Symfony\Component\HttpKernel\Profiler\Profiler */
+        $profiler = $this->container->get('profiler');
+        $profiler->disable();
+
+        $profile = $profiler->loadProfile($token);
+        $queries = $profile->getCollector('db')->getQueries();
+
+        if (!isset($queries[$connectionName][$query])) {
+            return new Response('This query does not exist.');
+        }
+
```

Comment:
you should check for ``$query['explainable']`` here (if it is false, you don't have the original params in the storage, and so you cannot run the query again)